sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -582,6 +582,7 @@ class CodegenContext {
     case array: ArrayType => genComp(array, c1, c2) + " == 0"
     case struct: StructType => genComp(struct, c1, c2) + " == 0"
     case udt: UserDefinedType[_] => genEqual(udt.sqlType, c1, c2)
+    case NullType => "true"
```

Comment:
Is this required? Will it be covered by any test?

BTW, the value should be `false`.

Reply:
I found the test case, but the test case is not affected by the value we generate here since it is under `nullSafeCodeGen`. 

However, we should still return `false` when doing `null = null`

Reply:
Yea, codegen fails without this. I had originally made the value `false` but when i noticed the codegen for comparison (https://github.com/aray/spark/blob/cc2f3eca28ee6b9faa87853568205307567827cc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala#L606) returned `0`, I changed it to be consistent. Happy to change it back though.